### PR TITLE
Move python venv to ./venv dir, add backup programming command

### PR DESCRIPTION
**Pull request by hcallahan-lowrisc**
I'm not sure why the fusesoc --run command does not work to program the hardware for me during testing, but manually reaching in and running the generated 'make pgm' command does work, so add it as a commented suggestion.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,7 +1,3 @@
-bin/
 build/
-lib/
-lib64
-pyvenv.cfg
-share/
 sw/build/
+venv/
```
Comment (andreaskurth) [inline .gitignore:3]: Maybe we would prefer making this a hidden directory (`.venv`), which [is common](https://docs.python.org/3/tutorial/venv.html#creating-virtual-environments)?

File: README.md [hunk 2]
```diff
@@ -50,11 +50,11 @@ First the software must be built. This is provide an initial binary for the FPGA
 build.
 
 ```
-cd sw
-mkdir build
-cd build
+mkdir sw/build
+pushd sw/build
```
Comment (marnovandermaas) [inline README.md:54]: Nice change!

File: README.md [hunk 3]
```diff
@@ -96,6 +96,9 @@ To program the FPGA, either use FuseSoC again
 
 ```
 fusesoc --cores-root=. run --target=synth --run lowrisc:ibex:demo_system
+
+# If the above does not work, try executing the programming operation manually with..
+make -C ./build/lowrisc_ibex_demo_system_0/synth-vivado/ pgm
```
Comment (marnovandermaas) [inline README.md:101]: Nice, I had found this was broken for me as well. I've been programming using the Vivado GUI. Also, we should be able to use the openFPGALoader, which I have vendored in with this PR: https://github.com/lowRISC/ibex-demo-system/pull/20

**Review by marnovandermaas — APPROVED** [review]
LGTM

File: .gitignore (hunk 1 above)
Comment (hcallahan-lowrisc) [reply .gitignore]: Good suggestion! done

**Pull request merged**